src/site/content/en/blog/mediastreamtrack-insertable-media-processing/index.md

```diff
@@ -0,0 +1,259 @@
+(For details of this type of process, see, for example,
+[video + canvas = magic](https://html5doctor.com/video-canvas-magic/).)
+
+## Use cases
+
+Use cases for insertable streams for `MediaStreamTrack` include, but are not limited to:
+
+- Video conferencing gadgets like "funny hats" or virtual backgrounds.
+- Voice processing like software [vocoders](https://en.wikipedia.org/wiki/Vocoder).
+
+## Current status {: #status }
+
+<div class="w-table-wrapper">
+
+| Step                                     | Status                   |
+| ---------------------------------------- | ------------------------ |
+| 1. Create explainer                      | [Complete][explainer]    |
+| 2. Create initial draft of specification | [In Progress][spec]      |
+| 3. Gather feedback & iterate on design   | [In progress](#feedback) |
+| 4. **Origin trial**                      | **[In progress][ot]**    |
+| 5. Launch                                | Not started              |
+
+</div>
+
+## How to use insertable streams for `MediaStreamTrack`
+
+### Enabling support during the origin trial phase
+
+Starting in Chrome 90, insertable streams for `MediaStreamTrack` is available as part of the WebCodecs origin trial in Chrome. The origin trial is expected to end in Chrome 91 (July 14, 2021). 
+If necessary, a separate origin trial will continue for insertable streams for `MediaStreamTrack`.
+
+{% include 'content/origin-trials.njk' %}
+
+### Register for the origin trial {: #register-for-ot }
+
+{% include 'content/origin-trial-register.njk' %}
+
+### Enabling via chrome://flags
+
+To experiment with insertable streams for `MediaStreamTrack` locally, without an origin trial token,
+enable the `#enable-experimental-web-platform-features` flag in `chrome://flags`.
+
+### Feature detection
+
+You can feature-detect insertable streams for `MediaStreamTrack` support as follows.
+
+```js
+if ('MediaStreamTrackProcessor' in window && 'MediaStreamTrackGenerator' in window) {
+  // Insertable streams for `MediaStreamTrack` is supported.
+}
+```
+
+### Core concepts
+
+Insertable streams for `MediaStreamTrack` builds on concepts previously proposed by
+[WebCodecs](https://web.dev/webcodecs/) and conceptually splits the `MediaStreamTrack` into two
+components:
+
+- The `MediaStreamTrackProcessor`, which consumes a `MediaStreamTrack` object's source and generates a stream
+  of media frames, specifically [`VideoFrame`](https://w3c.github.io/webcodecs/#videoframe-interface) or
+  [`AudioFrame`](https://w3c.github.io/webcodecs/#audioframe-interface)) objects. You can think of this as a
+  track sink that is capable of exposing the unencoded frames from the track as a `ReadableStream`.
+  It also exposes a control channel for signals going in the opposite direction.
+- The `MediaStreamTrackGenerator`, which consumes a stream of media frames and exposes a
+   `MediaStreamTrack` interface. The `MediaStreamTrackGenerator` can be provided to any sink,
```

Comment:
```suggestion
- The `MediaStreamTrackGenerator` consumes a stream of media frames and exposes a
   `MediaStreamTrack` interface. It can be provided to any sink,
```